### PR TITLE
#issue-178 Add CRON Group Generation action

### PR DESCRIPTION
**Pull request by serhiyzhovnir**
**Description** (*)
This PR adds Cron Group generation action.
Screenshot 2020-05-15 at 12 55 50
Screenshot 2020-05-15 at 12 56 12
Screenshot 2020-05-15 at 12 56 37

**Fixed Issues (if relevant)**
1. magento/magento2-phpstorm-plugin#178: Action/Code Generation. New cron group generation

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewCronGroupDialog.form [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+            <properties>
+              <text value="Add Option"/>
+            </properties>
+          </component>
+          <component id="f1e69" class="javax.swing.JLabel">
+            <constraints>
+              <grid row="3" column="3" row-span="1" col-span="1" vsize-policy="0" hsize-policy="0" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
+            </constraints>
+            <properties>
+              <text value="Frequency (in minutes) that schedules are written to the cron_schedule table."/>
```
Comment (eduard13) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/NewCronGroupDialog.form:227]: Let's move all these static tests into message bundle (e.g. `common.cronGroup.description.scheduleFrequency`), even for now we support only english version, later on will be easier to make the internationalization of this module.

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewCronGroupDialog.form [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+            <properties>
+              <text value=""/>
+            </properties>
+          </component>
+          <component id="f94bd" class="javax.swing.JLabel">
+            <constraints>
+              <grid row="0" column="0" row-span="1" col-span="1" vsize-policy="0" hsize-policy="0" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
+            </constraints>
+            <properties>
+              <text resource-bundle="magento2/common" key="common.cronGroup"/>
```
Comment (eduard13) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/NewCronGroupDialog.form:245]: Most probably that it makes sense to use `common.name`. What do you think?

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewCronGroupDialog.form (hunk 1 above)
Comment (eduard13) [reply src/com/magento/idea/magento2plugin/actions/generation/dialog/NewCronGroupDialog.form:227]: Also, is it possible to not allow to enter a negative value into dropdown? I know that there is a validation, but it doesn't make sense to allow to enter a negative value, and later on, to say that is not a valid value? This question it reflects to all dropdowns.

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/NewCronGroupDialog.java [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+        return addUseSeparateProcess.isSelected();
+    }
+
+    @SuppressWarnings({"PMD.UnusedPrivateMethod"})
+    private void createUIComponents() {
+        this.useSeparateProcess = new FilteredComboBox(getYesNoOptions());
+    }
+
+    private List<String> getYesNoOptions() {
+        return new ArrayList<>(Arrays.asList("Yes", "No"));
```
Comment (eduard13) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/NewCronGroupDialog.java:264]: Lets move these 2 values into common bundle. Please update everywhere where it's used.

**Comment by serhiyzhovnir** [comment]
Hi @eduard13
I've added the required changes.
About the validation for negative validation, I don't see an ability to restrict negative values for the numbers fields via component properties. 
I think that we can go as it is with validation for negative numbers, but please, let me if you have any ideas regarding it.
Thank you!

**Comment by eduard13** [comment]
Hi @serhiyzhovnir, thank you for fixes. Indeed there is no way to do this via component properties, however, we may set our own `SpinnerNumberModel` that allows to define the range and the increment.
I've already added the limitations and lets wait for the tests, after that will approve.

Thank you.

**Pull request merged**